### PR TITLE
Refactor gwpy.astro.inspiral_range to sensemon_range

### PR DESCRIPTION
**Pull request by alurban**
This PR refactors two functions under `gwpy.astro` to more appropriate namespaces:

* `inspiral_range --> sensemon_range`
* `inspiral_range_psd --> sensemon_range_psd`

This also includes a number of minor enhancements for efficiency.

The old functions are now placeholders which (via decoration) raise a `DeprecationWarning` so they can be replaced by more refined models that incorporate cosmological corrections in gwpy-2.2.0. This is related to #1292 and should be merged **well before** #1284.

### CODE REVIEW

File: gwpy/astro/range.py [hunk 1]
```diff
@@ -21,22 +21,48 @@
...
 from math import pi
+from scipy.integrate import trapz
 
-from scipy import integrate
-
-from astropy import (units, constants)
+from astropy import (
+    units,
+    constants,
+)
```
Comment (duncanmmacleod) [inline gwpy/astro/range.py:32]: Please reinstate the blank line between the standard library imports and the `scipy` import.

File: gwpy/astro/range.py [hunk 2]
```diff
@@ -75,9 +101,31 @@ def decorated_func(psd, *args, **kwargs):
...
+            "this estimator currently uses the LIGO SenseMon (LIGO-T030276), "
+            "which will be replaced by a model with cosmological corrections "
+            "in gwpy-2.2.0.\n\n"
+            "The proper SenseMon method is now gwpy.astro.{}".format(
+                func.__name__),
+            DeprecationWarning,
+        )
+        target.__doc__ = func.__doc__
+        return target
+    return decorated_func
```
Comment (duncanmmacleod) [inline gwpy/astro/range.py:117]: This looks like a reimplementation of the `gwpy.utils.decorators.deprecated_function` decorator, please consider using that instead.

File: gwpy/astro/range.py [hunk 3]
```diff
@@ -180,46 +215,46 @@ def inspiral_range(psd, snr=8, mass1=1.4, mass2=1.4, fmin=None, fmax=None,
...
-    mass2 = units.Quantity(mass2, 'solMass').to('kg')
-    mtotal = mass1 + mass2
-
-    # compute ISCO
-    fisco = (constants.c ** 3 / (constants.G * 6**1.5 * pi * mtotal)).to('Hz')
-
+    fisco = _get_isco_frequency(mass1, mass2)
     # format frequency limits
-    fmax = units.Quantity(fmax or fisco, 'Hz')
+    fmin = units.Quantity(fmin or psd.df.value, 'Hz')  # avoid DC value
```
Comment (duncanmmacleod) [inline gwpy/astro/range.py:233]: I think we should still be using `psd.df` here in case the units aren't `Hz`.

**Review by duncanmmacleod — CHANGES_REQUESTED** [review]
This looks good, and should set things up for #1284 nicely, however, one comment (possibly requiring a change) and one likely change required.

File: gwpy/astro/range.py (hunk 3 above)
Comment (alurban) [reply gwpy/astro/range.py]: Turns out you were right about this, this has been fixed in the latest commit

File: gwpy/astro/range.py (hunk 2 above)
Comment (alurban) [reply gwpy/astro/range.py]: Aha! This has been addressed, #1294 will need to go in first

File: gwpy/astro/range.py (hunk 1 above)
Comment (alurban) [reply gwpy/astro/range.py]: Done

**Comment by alurban** [comment]
@duncanmmacleod, this one is now ready and rebased against #1294

**Review by duncanmmacleod — APPROVED** [review]
This is ready.

**Pull request merged**